packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -84,4 +93,13 @@ class CupertinoPageScaffold extends StatelessWidget {
       ),
     );
   }
-}
\ No newline at end of file
+}
+
+/// Widget that fits in [CupertinoPageScaffold.navigationBar].
+///
+/// Must report its preferred size and whether it's opaque or translucent.
+abstract class BaseCupertinoNavigationBar extends PreferredSizeWidget {
```

Comment:
It'd be really nice to avoid this. It feels specific enough that I can't think of a great generalised class name. It would be nice to avoid tying it too closely to `CupertinoNavigationBar` and instead tie it to the underlying behaviour that it represents -- are we *completely* hiding what's below (in which case padding is applied one way) or not, in which case it's applied another.

Reply:
Done